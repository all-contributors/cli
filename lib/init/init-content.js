'use strict';

var _ = require('lodash/fp');
var injectContentBetween = require('../util').markdown.injectContentBetween;

var badgeContent = '[![All Contributors](https://img.shields.io/badge/all_contributors-0-orange.svg?style=flat-square)](#contributors)';
var headerContent = 'Thanks goes to these wonderful people ([emoji key](https://github.com/kentcdodds/all-contributors#emoji-key)):';
var listContent = '<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section --><!-- ALL-CONTRIBUTORS-LIST:END -->';
var footerContent = 'This project follows the [all-contributors](https://github.com/kentcdodds/all-contributors) specification. Contributions of any kind welcome!';

function addBadge(lines) {
  return injectContentBetween(lines, badgeContent, 1, 1);
}

function splitAndRejoin(fn) {
  return _.flow(
    _.split('\n'),
    fn,
    _.join('\n')
  );
}

var findContributorsSection = _.findIndex(function isContributorsSection(str) {
  return str
    .toLowerCase()
    .indexOf('# contributors') === 1;
});

function addContributorsList(lines) {
  var insertionLine = findContributorsSection(lines);
  if (insertionLine === -1) {
    return lines
      .concat([
        '## Contributors',
        '',
        headerContent,
        '',
        listContent,
        '',
        footerContent
      ]);
  }
  return injectContentBetween(lines, listContent, insertionLine + 2, insertionLine + 2);
}

module.exports = {
  addBadge: splitAndRejoin(addBadge),
  addContributorsList: splitAndRejoin(addContributorsList)
};
