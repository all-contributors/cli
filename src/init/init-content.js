const _ = require('lodash/fp')
const injectContentBetween = require('../util').markdown.injectContentBetween

const badgeContent = [
  '<!-- ALL-CONTRIBUTORS-BADGE:START - Do not remove or modify this section -->',
  '[![All Contributors](https://img.shields.io/badge/all_contributors-0-orange.svg?style=flat-square)](#contributors-)',
  '<!-- ALL-CONTRIBUTORS-BADGE:END -->',
].join('\n')

const headerContent =
  'Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/docs/en/emoji-key)):'
const listContent = [
  '<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->',
  '<!-- prettier-ignore-start -->',
  '<!-- markdownlint-disable -->',
  '<!-- markdownlint-restore -->',
  '<!-- prettier-ignore-end -->',
  '<!-- ALL-CONTRIBUTORS-LIST:END -->',
].join('\n')
const footerContent =
  'This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!'

function addBadge(lines) {
  return injectContentBetween(lines, badgeContent, 1, 1)
}

function splitAndRejoin(fn) {
  return _.flow(
    _.split('\n'),
    fn,
    _.join('\n'),
  )
}

const findContributorsSection = _.findIndex(function isContributorsSection(
  str,
) {
  return str.toLowerCase().indexOf('# contributors') === 1
})

function addContributorsList(lines) {
  const insertionLine = findContributorsSection(lines)
  if (insertionLine === -1) {
    return lines.concat([
      '## Contributors âœ¨',
      '',
      headerContent,
      '',
      listContent,
      '',
      footerContent,
    ])
  }
  return injectContentBetween(
    lines,
    listContent,
    insertionLine + 3,
    insertionLine + 3,
  )
}

module.exports = {
  addBadge: splitAndRejoin(addBadge),
  addContributorsList: splitAndRejoin(addContributorsList),
}
