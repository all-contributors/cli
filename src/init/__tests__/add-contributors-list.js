import {unlink} from 'fs'
import {addContributorsList} from '../init-content'
import ensureFileExists from '../file-exist'

test('insert list under contributors section', () => {
  const content = [
    '# project',
    '',
    'Description',
    '',
    '## Contributors',
    '',
  ].join('\n')
  const result = addContributorsList(content)

  expect(result).toMatchSnapshot()
})

test('create contributors section if it is absent', () => {
  const content = ['# project', '', 'Description'].join('\n')
  const result = addContributorsList(content)

  expect(result).toMatchSnapshot()
})

test('create contributors section if content is empty', () => {
  const content = ''
  const result = addContributorsList(content)

  expect(result).toMatchSnapshot()
})

test('README exists', done => {
  const file = 'README.md'
  ensureFileExists(file)
    .then(data => expect(data).toStrictEqual(file))
    .then(_ => done())
})

test("LOREM doesn't exists", done => {
  const file = 'LOREM.md'
  ensureFileExists(file).then(data => {
    expect(data).toStrictEqual(file)
    return unlink(file, err => {
      if (err) throw err
      done()
    })
  })
})
