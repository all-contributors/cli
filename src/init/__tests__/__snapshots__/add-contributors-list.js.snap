// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create contributors section if content is empty 1`] = `
"
## Contributors

Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/docs/en/emoji-key)):

<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
<!-- prettier-ignore -->
<!-- ALL-CONTRIBUTORS-LIST:END -->

This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!"
`;

exports[`create contributors section if it is absent 1`] = `
"# project

Description
## Contributors

Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/docs/en/emoji-key)):

<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
<!-- prettier-ignore -->
<!-- ALL-CONTRIBUTORS-LIST:END -->

This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!"
`;

exports[`insert list under contributors section 1`] = `
"# project

Description

## Contributors

<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
<!-- prettier-ignore -->
<!-- ALL-CONTRIBUTORS-LIST:END -->"
`;
